store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -365,8 +365,13 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
             }
         }
 
-        AuthToken at = null;
-        at = expires ==  0 ? AuthProvider.getAuthToken(acct, tokenType) : AuthProvider.getAuthToken(acct, expires, tokenType);
+        AuthToken at = expires ==  0 ? AuthProvider.getAuthToken(acct, tokenType) : AuthProvider.getAuthToken(acct, expires, tokenType);
+        if (registerTrustedDevice && (trustedToken == null || trustedToken.isExpired())) {
+            //generate a new trusted device token if there is no existing one or if the current one is no longer valid
+            Map<String, Object> attrs = getTrustedDeviceAttrs(zsc, newDeviceId == null? deviceId: newDeviceId);
+            TrustedDevices trustedDeviceManager = TwoFactorAuth.getFactory().getTrustedDevices(acct);
+            trustedToken = trustedDeviceManager.registerTrustedDevice(attrs);
```

Comment:
Do we need to handle **trustedDeviceManager** being null?

Reply:
added null check.